### PR TITLE
Add some initial documentation for univeral algebras. closes #120

### LINKED ISSUE #120: Document SemigroupK and MonoidK
I've already seen several people ask about the difference between `Semigroup` and `SemigroupK`. These "universal" type classes that work at the kind level instead of the type level should be documented.

There may be others besides SemigroupK and MonoidK - these are the ones that come to mind.

For now I think we should just document these in the ScalaDoc for the type class traits. Eventually this may be something we add in tut-generated documentation.

### PR DESCRIPTION
**Pull request by non**
Fixes #120

### CODE REVIEW

File: core/src/main/scala/cats/MonoidK.scala [hunk 1]
```diff
@@ -2,9 +2,36 @@ package cats
...
+ */
 @typeclass trait MonoidK[F[_]] extends SemigroupK[F] { self =>
+
+  /**
+   * Given a type A, create an "empty" F[A] value.
+   */
   def empty[A]: F[A]
 
+  /**
+   * Given a type A, create a concrete Semigroup[F[A]].
```
Comment (fthomas) [inline core/src/main/scala/cats/MonoidK.scala]: Should be `Monoid[F[A]]`, right?

File: core/src/main/scala/cats/SemigroupK.scala [hunk 2]
```diff
@@ -2,14 +2,42 @@ package cats
...
 
+/**
+ * SemigroupK is a universal semigroup which operates on kinds.
+ * 
+ * This type class is useful when its type parameter F[_] has a
+ * structure that can be combined for any particular type. Thus,
+ * SemigroupK is like a Semigroup for kinds (i.e. parameterized
+ * types).
+ * 
+ * A Semigroup[F] can produce a Semigroup[F[A]] for any type A.
```
Comment (fthomas) [inline core/src/main/scala/cats/SemigroupK.scala]: Should be `SemigroupK[F]` instead of `Semigroup[F]`.

Comment (non) [inline core/src/main/scala/cats/SemigroupK.scala]: Yes, thanks for catching that!

File: core/src/main/scala/cats/MonoidK.scala (hunk 1 above)
Comment (non) [inline core/src/main/scala/cats/MonoidK.scala]: Yes, thanks!

**Comment by ceedubs** [comment]
Thanks @non. This looks really good to me.

I have one minor wording note. Both places that you have "for any particular type" I would expect something more like "for all arbitrary types" or something. To me the former sounds parallel with "exists" and the latter with "forall". English is weird though, and others might not read it the same way I did.

**Comment by non** [comment]
Hm. Maybe I'm just a pedantic ex-philosopher but I hear "for *any* particular type" as `forall` and "for *a* particular type" as `exists`.

Maybe I should change this to "no matter what type A is" or something?

**Comment by ceedubs** [comment]
@non with all of the context you give, I don't think it would trip people up anyway. I say :+1: and if we find out wording is confusing people we can change it.

**Comment by stew** [comment]
:+1:

**Pull request merged**